queue/sqs/reader.go

```
@@ -203,22 +203,25 @@ func (r *Reader) processAndTrack(msg *sqs.Message, token interface{}) {
 		atomic.AddUint32(&r.nProcessingMessages, ^uint32(0))
 		r.wg.Done()
 	}()
-	if msg.Body == nil {
-		r.log.Errorf("unexpected empty body message from sqs")
+	err := validateSQSMessage(msg)
+	if err != nil {
+		r.log.Errorf("error %+v", err)
```

If we get to this point, and `msg` happens to be nil, then I think we should simply return the error generated by `validateSQSMessage`, otherwise we are going to `panic` at line `209`. What do you think?

---

Yeah I though about it, theoretically msg can not be nil, but it's true that is not clearly stated anywhere, so what I'm going to do it's to check for the msg not being nil at the begging of the method, because msg being different of nil is a precondition of the function.

---

That's great! Thanks @manelmontilla !